src/coreclr/nativeaot/System.Private.Reflection.Execution/src/Internal/Reflection/Execution/MethodInvokers/InstanceMethodInvoker.cs

```diff
@@ -74,5 +106,13 @@ public sealed override Delegate CreateDelegate(RuntimeTypeHandle delegateType, o
         public sealed override IntPtr LdFtnResult => MethodInvokeInfo.LdFtnResult;
 
         private RuntimeTypeHandle _declaringTypeHandle;
+        private delegate*<nint, object> _allocatorMethod;
+
+        private static class RawCalliHelper
+        {
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            public static T Call<T>(delegate*<IntPtr, T> pfn, IntPtr arg)
+            => pfn(arg);
+        }
```

Comment:
FYI @yowl; this will need to be adjusted to use the same calling convention transform we do for `Activator.CreateInstance` in the next integration.

Reply:
> FYI @yowl; this will need to be adjusted to use the same calling convention transform we do for `Activator.CreateInstance` in the next integration.

Is the issue that I used `IntPtr` instead of `MethodTable*`? Otherwise this should be just a "regular" calli. It's in a RawCalliHelper to avoid the fat pointer penalty (we know the target doesn't need a generic context) but otherwise just any other user calli.

Reply:
No, the issue is the mismatch of calling conventions (equivalently, the assumption that they are the same between managed and native). This is a managed `calli`, so it will get the shadow stack argument which the native allocator methods do not have. [We solve this downstream via detouring through native code](https://github.com/dotnet/runtimelab/blob/588b1e37a5caeefd187bc8ddc2e4735c3da2c183/src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/TypeLoaderExports.cs#L373-L388). It could be solved upstream, but not sure there is appetite for that, additionally, I suspect we'll start using custom allocators that have the shadow stack argument sooner or later for performance reasons.

Reply:
Ah, I see, thanks and sorry for the extra trouble!